src/ifcwrap-csharp/IfcOpenShell.Net/IfcOpenShell.Net/PlacementExtensions.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.IO;
+using System.Numerics;
+using System.Numerics.Double;
+
+namespace IfcOpenShell
+{
+    public static class PlacementExtensions
+    {
+        // TODO: use the matrix4 type that ifcopenshell 0.8.0 introduced
+        /// <summary>
+        /// Parse a local placement into a 4x4 transformation matrix.
+        /// This is typically used to find the location and rotation of an element. The
+        ///     transformation matrix takes the form of:
+        ///
+        /// .. code::
+        ///
+        /// [ [ x_x, y_x, z_x, x   ]
+        /// [ x_y, y_y, z_y, y   ]
+        /// [ x_z, y_z, z_z, z   ]
+        /// [ 0.0, 0.0, 0.0, 1.0 ] ]
+        ///
+        /// Example:
+        ///
+        /// .. code:: python
+        ///
+        ///     placement = file.by_type("IfcBeam")[0].ObjectPlacement
+        /// matrix = ifcopenshell.util.placement.get_local_placement(placement)
+        ///
+        /// :param placement: The IfcLocalPlacement entity
+        /// :type placement: ifcopenshell.entity_instance, optional
+        /// :return: A 4x4 numpy matrix
+        /// :rtype: MatrixType
+        /// </summary>
+        /// <param name="placement">The placement of the entity</param>
+        /// <returns></returns>
+        public static DoubleMatrix4x4 GetLocalPlacement(EntityInstance placement)
+        {
+            if (placement == null)
+            {
+                return DoubleMatrix4x4.Identity;
+            }
+
+            DoubleMatrix4x4 parent;
+            if (!placement.TryGetAttributeAsEntity("PlacementRelTo", out var relTo))
+            {
+                parent = DoubleMatrix4x4.Identity;
+            }
+            else
+            {
+                parent = GetLocalPlacement(relTo);
+            }
+            if(placement.TryGetAttributeAsEntity("RelativePlacement", out var relativePlacement))
+            {
+                return DoubleMatrix4x4.Multiply(parent,
+                    GetAxis2Placement(relativePlacement));
+            }
+            // this case probably cannot happen, as it does not exist in the python implementation
+            return parent;
+        }
+        
+        // TODO: Double precision matrix / vector implementations :crying:
+        public static DoubleMatrix4x4 GetAxis2Placement(EntityInstance placement)
+        {
+            if (placement == null)
+            {
+                return DoubleMatrix4x4.Identity;
+            }
+
+            DoubleVector3 x = DoubleVector3.UnitX;
+            DoubleVector3 z = DoubleVector3.UnitZ;
+            DoubleVector3 o = DoubleVector3.UnitY;
+            
+            
+            var ifcClass = placement.Is();
+            switch (ifcClass)
```

Comment:
Add a comment that his is copyied from python and rerference the file it was copied from